### PR TITLE
Action Manager | Menu Manager Add Widget API

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Introduces an API to add widgets to menus in the new Action Manager.

image

This API bypasses actions and allows to add a QWidget* directly to a menu. This is because with the current implementation, only one instance of the widget can be displayed at a time - multiple usages of the underlying WidgetAction object would result in the widget only being shown in the first location it is called, which would not work well with the concept of Actions being reusable in multiple places.

Ideally it would be possible to create an action embedding a widget, but the caller would have to implement a factory function. There's no real use case for this right now (all cases I have seen throughout the Editor so far only added the widget to a single menu/toolbar, and we don't have a concept of cloning menus/toolbars in the API) but if the needs change, this could be an option that gets explored.

This PR also
- Migrates the "Search documentation..." search box in the Help menu from the legacy action manager to the new one;
- Fixes an issue with the widget mentioned above, which would open a url in the browser with the search results, but would point to the wrong page resulting in the search not being displayed.

This also adds a couple of unit tests to verify the basic functionality.

### CODE REVIEW

File: Code/Editor/Core/LevelEditorMenuHandler.cpp [hunk 1]
```diff
@@ -758,24 +758,24 @@ QMenu* LevelEditorMenuHandler::CreateHelpMenu()
...
+    lineEdit->setPlaceholderText(tr("Search documentation..."));
     lineEditSearchAction->setDefaultWidget(containerWidget);
 
     auto searchAction = [lineEdit]()
         {
             auto text = lineEdit->text();
             if (text.isEmpty())
             {
-                QDesktopServices::openUrl(QUrl("https://o3de.org/docs/"));
+                QDesktopServices::openUrl(QUrl("https://www.o3de.org/docs/"));
```
Comment (pollend) [inline Code/Editor/Core/LevelEditorMenuHandler.cpp:769]: is the `www` necessary?

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.cpp [hunk 2]
```diff
@@ -80,6 +86,11 @@ namespace AzToolsFramework
                     m_menu->addSeparator();
                     break;
                 }
+            case MenuItemType::Widget:
+                {
+                    m_menu->addAction(elem.second.m_widgetAction);
+                    break;
+                }
```
Comment (pollend) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.cpp:93]: for some of these cases are the braces necessary.

**Review by pollend — APPROVED** [review]

File: Code/Editor/Core/LevelEditorMenuHandler.cpp (hunk 1 above)
Comment (AMZN-daimini) [reply Code/Editor/Core/LevelEditorMenuHandler.cpp:769]: It is not necessary, but it is the correct url (trying to access `https://o3de.org/` will redirect to `https://www.o3de.org/`). The real fix here is on the line below, switching from `/docs/` to /`search/`.

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.cpp (hunk 2 above)
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Menu/EditorMenu.cpp:93]: They are not necessary for compilation, but I tend to add them as an extension of the curly brackets policy in the style guide
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#use-of-curly-braces
(It is not listed and the switch without brackets is considered acceptable, but I tend to use them anyways as a habit)

File: Code/Editor/Core/LevelEditorMenuHandler.cpp [hunk 3]
```diff
@@ -758,24 +758,24 @@ QMenu* LevelEditorMenuHandler::CreateHelpMenu()
...
             }
             else
             {
                 static const int versionStringSize = 128;
                 char productVersionString[versionStringSize];
                 const SFileVersion& productVersion = gEnv->pSystem->GetProductVersion();
                 productVersion.ToString(productVersionString, versionStringSize);
-
-                QUrl docSearchUrl("https://o3de.org/docs/");
+
```
Comment (cgalvan) [inline Code/Editor/Core/LevelEditorMenuHandler.cpp:777]: Looks like some tab space got inserted here accidentally?

File: Code/Editor/MainWindow.cpp [hunk 4]
```diff
@@ -12,14 +12,17 @@
...
 // Qt
-#include <QMenuBar>
-#include <QDebug>
-#include <QMessageBox>
-#include <QInputDialog>
-#include <QHBoxLayout>
 #ifdef Q_OS_WIN
 #include <QAbstractEventDispatcher>
 #endif
+#include <QDebug>
```
Comment (cgalvan) [inline Code/Editor/MainWindow.cpp:18]: While you're re-organizing these, I think we can just remove the `QDebug` include (it doesn't look to be used)

**Review by cgalvan — APPROVED** [review]

File: Code/Editor/MainWindow.cpp [hunk 5]
```diff
@@ -2429,24 +2433,68 @@ void MainWindow::InitializeMenus()
...
+        auto searchAction = [lineEdit]()
+        {
+            auto text = lineEdit->text();
+            if (text.isEmpty())
+            {
+                QDesktopServices::openUrl(QUrl("https://www.o3de.org/docs/"));
+            }
+            else
+            {
+                static const int versionStringSize = 128;
```
[the remark this answers is not in the document]
Comment (AMZN-daimini) [reply Code/Editor/MainWindow.cpp:2456]: I was merely migrating the old code to a new place, so I'm not sure. That whole string doesn't seem to be used in the url generation anyways, so I removed it altogether :)

File: Code/Editor/MainWindow.cpp (hunk 4 above)
Comment (AMZN-daimini) [reply Code/Editor/MainWindow.cpp:18]: Done!

File: Code/Editor/Core/LevelEditorMenuHandler.cpp (hunk 3 above)
Comment (AMZN-daimini) [reply Code/Editor/Core/LevelEditorMenuHandler.cpp:777]: Looks fine on my machine, not sure why the diff shows it like this :D

Comment (AMZN-daimini) [reply Code/Editor/Core/LevelEditorMenuHandler.cpp]: Ok the new submission fixed the diff :)

**Pull request merged**